app/user.go

```diff
@@ -155,8 +155,10 @@ func (a *App) CreateUserWithInviteId(user *model.User, inviteId, redirect string
 
 	a.AddDirectChannels(team.Id, ruser)
 
-	if err := a.Srv().EmailService.sendWelcomeEmail(ruser.Id, ruser.Email, ruser.EmailVerified, ruser.Locale, a.GetSiteURL(), redirect); err != nil {
-		mlog.Warn("Failed to send welcome email on create user with inviteId", mlog.Err(err))
+	if !user.DisableWelcomeEmail {
```

Comment:
I'd assign the `DIsableWelcomeEmail` value to the new `ruser` and make the check inside the `sendWelcomeEmail` method to avoid adding conditionals every time we want to check